web3/middleware/gas_price.py

```diff
@@ -0,0 +1,18 @@
+from cytoolz.dicttoolz import (
+    assoc,
+)
+
+
+def gas_price_middleware(make_request, web3):
+    """
+    Includes a gas price using the gas price strategy
+    """
+    def middleware(method, params):
+        if method == 'eth_sendTransaction':
+            transaction = params[0]
+            generated_gas_price = web3.eth.generateGasPrice(transaction)
+            if 'gasPrice' not in transaction and generated_gas_price:
```

Comment:
I think the check should be explicitly `... and generated_gas_price is not None` to allow the subtle edge case of returning `0` as the gas price which while strange, should be a valid response.